### PR TITLE
compatibility with other rmw implementations

### PR DESCRIPTION
**Pull request by dirk-thomas**
Connects to ros2/ros2#42

@esteve @tfoote @wjwwood Please review.

### CODE REVIEW

**Comment by esteve** [comment]
+1

File: rmw_connext_dynamic_cpp/src/functions.cpp [hunk 1]
```diff
@@ -364,7 +373,6 @@ void _publish(DDS_DynamicData * dynamic_data, const void * ros_message,
   //DDS_DynamicData * dynamic_data = ddts->create_data();
   for (unsigned long i = 0; i < members->member_count_; ++i) {
     const rosidl_typesupport_introspection_cpp::MessageMember * member = members->members_ + i;
-    DDS_TypeCode * member_type_code;
```
Comment (tfoote) [inline rmw_connext_dynamic_cpp/src/functions.cpp]: Is this just unrelated cleanup?

Comment (dirk-thomas) [inline rmw_connext_dynamic_cpp/src/functions.cpp]: Yeah, that variable was not used anywhere.

**Comment by tfoote** [comment]
+1
